### ISSUE TITLE
Compare and contrast with Shen and Lux

### ISSUE THREAD

**Issue by yurrriq** [ox-lang/ox#34]
I'm curious how Ox compares to [Shen](http://shenlanguage.org) and [Lux](https://github.com/LuxLang/lux).

**Comment by yurrriq** [comment]
Also what's the deal with [Jaunt](https://github.com/jaunt-lang/jaunt)?

**Comment by arrdem** [comment]
I should really put this in a FAQ section somewhere.

## Ox vs Clojure

Ahahaha. No.

## Ox vs Haskell

Ox is by a lone amateur. Haskell has a strong established community and full-time maintainers. I'm working on Ox for fun, and because maybe some day it'll solve some friction points that seem to chase me around.

## Ox vs Shen

Shen gets a lot right in terms of having a very very small kernel and porting well to multiple implementations. I'm very impressed with Shen, as am I with Scheme. But is marred by a history of weirdness, Dr. Taver's own idiosyncrasies, a rather badly written book and lacks a "flagship" implementation one would actually turn to for meaningful deployment besides Dr. Taver's CL backend. I'm sure @deech would be delighted to dispute this assessment.

## Ox vs Lux

Both are largely headed in the same direction. On cursory inspection, Ox is doing stuff with integrating packages and versioning into the language which Lux seems to have no truck with. Lux seems to have strayed farther from the Scheme notation tradition than I intend to in various ways. I'll have to take a more meaningful look here.

## Ox vs Jaunt

Jaunt is dead. The carrying cost of moving Clojure's codebase and legacy is too high. Clojure compatibility is an exercise in futility. A clean break is emotionally easier for me to sustain and offers a wide range of interesting avenues which mesh directly with my now fairly well formed ideas around what I want to build.

## Ox vs Kiss/Magic
- Kiss: github.com/mikera/kiss
- Magic: github.com/mikera/magic

Again, very similar ideas and direction. Mike and I have corresponded a fair bit. Mike is more interested in a near Clojure language than I am, and his view of the place of types in a language is more in-line with the JVM's interpretation of types while mine is more informed by Haskell. If either of us finish we'll wind up in largely the same niche at least on the JVM but I think that Magic/Kiss will deliberately smell like a clean-room Clojure-but-again whereas Ox is trying to be something else.

**Comment by yurrriq** [comment]
Thanks for this. :+1: to a FAQ. P.S. My friend made me read it aloud at the bar. I was too excited to ignore the notification and she was too "quit playing on your phone" to let me not share.

**Comment by deech** [comment]
You're not wrong about the "flagship",  the reference SBCL implementation isn't a finished product by any means. I prefer to think of it as a prototype. 

I would disagree that the book is badly written. The typesetting could definitely use some work but the content is strong, better in my opinion than most programming texts. That said there is certainly a place for more documentation.

As to Tarver, he's is working on his own proprietary Shen implementation and isn't much involved with OSS part and hasn't been for a couple of years. You can safely assume that aside from his quirks, whatever they may be, won't impeded it's progress. On the contrary he has been pretty responsive to questions on the newsgroup.

**Comment by mikera** [comment]
FWIW I've merged all Kiss development into Magic, which will be the language main project I am working on going forward.

Regarding the Type system question: I'm actually more interested in an Idris style type system, *while maintaining compatibility* with the host / JVM notion of types, at least for the purposes of fast dispatch and interop.  This seems feasible and potentially quite elegant from my current (naive?) perspective.
